File: 20200809/a.cpp
#include <iostream>
#include <cmath>
using namespace std;
bool is_integer(double x)
{
  return floor(x) == ceil(x);
}
int main()
{
  int n;
  cin >> n;
  long double tmp[200000];
  int a[200000];
  for (int i = 0; i < n; i++)
  {
    cin >> a[i];
  }

  int cnt = 0;
  for (int i = 0; i < n; i++)
  {
    for (int j = i + 1; j < n; j++)
    {
      if (is_integer(a[i] * a[j]))
      {
        cout << a[i] << ',' << a[j] << '=' << a[i] * a[j] << endl;
        cnt++;
      }
    }
  }
  cout << cnt << endl;
}

File: 20200809/a.py
import math
n = int(input())
a = []
for i in range(n):
    a.append(float(input()) * 10 ** 9)

cnt = 0
for i in range(n):
    for j in range(i + 1, n):
        if (((a[i] * a[j]) / 10 ** 18).is_integer()):
            cnt += 1
print(cnt)
